pkg/asset/cluster/azure/azure.go

```diff
@@ -12,3 +21,44 @@ func Metadata(config *types.InstallConfig) *azure.Metadata {
 		Region: config.Platform.Azure.Region,
 	}
 }
+
+// PreTerraform performs any infrastructure initialization which must
+// happen before Terraform creates the remaining infrastructure.
+func PreTerraform(ctx context.Context, clusterID *installconfig.ClusterID, installConfig *installconfig.InstallConfig) error {
+	clusterName := installConfig.Config.ObjectMeta.Name
+	azureDNS, _ := azureconfig.NewDNSConfig()
+	record := fmt.Sprintf("api.%s", clusterName)
+
+	exists := false
+	rgName := installConfig.Config.Azure.BaseDomainResourceGroupName
+	zoneName := installConfig.Config.BaseDomain
+
+	// Look for an existing CNAME first
+	rs, err := azureDNS.GetDNSRecordSet(rgName, zoneName, record, azdns.CNAME)
+	if err == nil && rs.CnameRecord != nil {
+		exists = true
```

Comment:
shouldn't we just return and report error here? why do we go ahead and again check for A, AAAA records

i think early return will be simpler here and also simplify later conditionals.